ios/FluentUI/Core/Colors2.swift

```diff
@@ -0,0 +1,243 @@
+        brandColors[.brandBackgroundTint] = brandBackgroundTint
+    }
+    if let brandBackgroundDisabled = provider.brandBackgroundDisabled(for: theme)?.dynamicColor {
+        brandColors[.brandBackgroundDisabled] = brandBackgroundDisabled
+    }
+    if let brandForeground1 = provider.brandForeground1(for: theme)?.dynamicColor {
+        brandColors[.brandForeground1] = brandForeground1
+    }
+    if let brandForeground1Pressed = provider.brandForeground1Pressed(for: theme)?.dynamicColor {
+        brandColors[.brandForeground1Pressed] = brandForeground1Pressed
+    }
+    if let brandForeground1Selected = provider.brandForeground1Selected(for: theme)?.dynamicColor {
+        brandColors[.brandForeground1Selected] = brandForeground1Selected
+    }
+    if let brandForegroundTint = provider.brandForegroundTint(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundTint] = brandForegroundTint
+    }
+    if let brandForegroundDisabled1 = provider.brandForegroundDisabled1(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundDisabled1] = brandForegroundDisabled1
+    }
+    if let brandForegroundDisabled2 = provider.brandForegroundDisabled2(for: theme)?.dynamicColor {
+        brandColors[.brandForegroundDisabled2] = brandForegroundDisabled2
+    }
+    if let brandStroke1 = provider.brandStroke1(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1] = brandStroke1
+    }
+    if let brandStroke1Pressed = provider.brandStroke1Pressed(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Pressed] = brandStroke1Pressed
+    }
+    if let brandStroke1Selected = provider.brandStroke1Selected(for: theme)?.dynamicColor {
+        brandColors[.brandStroke1Selected] = brandStroke1Selected
+    }
+    return brandColors
+}
+
+// MARK: Colors
+
+@objc(MSFColors2)
+public final class Colors2: NSObject {
+    /// Associates a `ColorProvider2` with a given `UIWindow` instance.
+    ///
+    /// - Parameters:
+    ///   - provider: The `ColorProvider2` whose colors should be used for controls in this window.
+    ///   - window: The window where these colors should be applied.
+    @objc public static func setProvider(provider: ColorProviding2, for theme: FluentTheme) {
+        colorProvidersMap.setObject(provider, forKey: theme)
+
+        // Create an updated fluent theme as well
+        let brandColors = brandColorOverrides(provider: provider, for: theme)
+        brandColors.forEach { token, value in
+            theme.aliasTokens.colors[token] = value
+        }
+    }
+
+    /// Removes any associated `ColorProvider2` from the given `UIWindow` instance.
+    ///
+    /// - Parameters:
+    ///   - window: The window that should have its `ColorProvider2` removed.
+    @objc public static func removeProvider(for theme: FluentTheme) {
+        colorProvidersMap.removeObject(forKey: theme)
+        // TODO: what is the equivalent function here?
+        // window.fluentTheme = FluentThemeKey.defaultValue
```

Comment:
Probably something like this:
```swift
brandColors.forEach { token, _ in
    theme.aliasTokens.colors[token] = nil
}
```

Reply:
Please see my updated attempt at resetting values - this does not yet work, but is conceptually close to what I'm intending to do (iterate through and reset the brand tokens to their original values)